xt/06.1-Perl-PO.t

```diff
@@ -0,0 +1,95 @@
+#!/usr/bin/perl
+
+=description
+
+This script scans various Perl files for translatable strings.
+
+The scanner uses the Perl Pattern Recognizer to find text() or marktext()
+translation routines, either in code or comments, because they sometimes
+intentionally contain translatable strings.
+
+This script exists because xgettext and xgettext.pl don't allow us to
+extract a sub-set of strings from our SQL files.
+
+=cut
+
+use strict;
+use warnings;
+
+use PPR;
+use utf8;
+use Test::More;
+use File::Find;
+
+my @on_disk;
+sub collect {
+    return if $File::Find::name !~ m/\.(pl|pm|t|tex)$/;
+
+    my $module = $File::Find::name;
+    return if $module =~ m/07.1-extract-perl.t/;
+    push @on_disk, $module
+}
+find(\&collect, 'lib/LedgerSMB/', 'old/lib/LedgerSMB/');
+
+plan tests => scalar @on_disk;
+
+sub slurp{ local (*ARGV, $/); @ARGV = shift; readline; }
+
+my @lines;        # To record line numbers
+my $line;
+
+for my $file (@on_disk) {
+
+    subtest $file => sub {
+        my $source = slurp($file);
+        my $errors = 0;
+
+        my (@lines,$line);  # Record line numbers
+        $line = 1;
+        my @text = grep {defined}
+            $source =~
+            m{
+               ((?&PerlEndOfLine) (??{ ++$line }))*?     # Count the lines
+               (?: (?:[Mm]ark)?[Tt]ext\b                 # Introducer
+                   (?&PerlOWS) [\(\s] (?&PerlOWS))       # '(' or ' '
+               ((?&PerlString)) (?{ push @lines,$line }) # string and line
+               $PPR::GRAMMAR                             # Preload our grammar
+             }gmx;
+
+        # Deduplicate, standardize and pack line references
+        my %text;
+        for (@text) {

```

Comment:
The content of the for block suggests that the extract-perl functionality isn't sufficiently abstracted into a library as this looks like a partial reimplementation.